lib/which_formula.rb

```diff
@@ -9,9 +9,7 @@ module WhichFormula
     module_function
 
     def matches(cmd)
-      # We use 'grep' here to speed up our search
-      # TODO: benchmark grep vs. reading the file line-by-line in Ruby
-      Utils.popen_read("grep", "--color=never", cmd, LIST_PATH).chomp.split("\n")
+      File.readlines(LIST_PATH).map(&:chomp).select { |line| line.include?(cmd) }
```

Comment:
```suggestion
      File.readlines(LIST_PATH).grep(cmd).map(&:chomp)
```
Maybe? Will need benchmarking again though.